hopsworks-api/src/main/java/io/hops/hopsworks/api/metadata/XAttrsResource.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * This file is part of Hopsworks
+ * Copyright (C) 2019, Logical Clocks AB. All rights reserved
+ *
+ * Hopsworks is free software: you can redistribute it and/or modify it under the terms of
+ * the GNU Affero General Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later version.
+ *
+ * Hopsworks is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ * PURPOSE.  See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ */
+package io.hops.hopsworks.api.metadata;
+
+import io.hops.hopsworks.api.filter.AllowedProjectRoles;
+import io.hops.hopsworks.api.filter.Audience;
+import io.hops.hopsworks.api.jwt.JWTHelper;
+import io.hops.hopsworks.common.dao.project.Project;
+import io.hops.hopsworks.common.dao.project.ProjectFacade;
+import io.hops.hopsworks.common.dao.user.Users;
+import io.hops.hopsworks.exceptions.DatasetException;
+import io.hops.hopsworks.exceptions.MetadataException;
+import io.hops.hopsworks.jwt.annotation.JWTRequired;
+import io.hops.hopsworks.restutils.RESTCodes;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import org.json.JSONObject;
+
+import javax.ejb.EJB;
+import javax.ejb.TransactionAttribute;
+import javax.ejb.TransactionAttributeType;
+import javax.enterprise.context.RequestScoped;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.SecurityContext;
+import javax.ws.rs.core.UriBuilder;
+import javax.ws.rs.core.UriInfo;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Level;
+
+@Api(value = "Extended Attributes Resource")
+@RequestScoped
+@TransactionAttribute(TransactionAttributeType.NEVER)
+public class XAttrsResource {
+  
+  @EJB
+  private ProjectFacade projectFacade;
+  @EJB
+  private JWTHelper jWTHelper;
+  @EJB
+  private XAttrsController xattrsController;
+  
+  private Project project;
+  
+  public void setProject(Integer projectId) {
+    this.project = projectFacade.find(projectId);
+  }
+  
+  
+  @ApiOperation( value = "Create or Update an extended attribute for a path.")
+  @PUT
+  @Path("{path: .+}")
```

Comment:
@ErmiasG Would that fail if we create a dataset with name `datasets`? As in the new datsets API we do `/project/{id}/datasets/{path: .+}` ?

Reply:
it is under @Path("{projectId}/xattrs"), so it should resolve to /project/{id}/xattrs/datasets